odbc_linux.c

```diff
@@ -0,0 +1,391 @@
+/*
+ * Copyright (C) 2008 Search Solution Corporation. All rights reserved by Search Solution. 
+ *
+ * Redistribution and use in source and binary forms, with or without modification, 
+ * are permitted provided that the following conditions are met: 
+ *
+ * - Redistributions of source code must retain the above copyright notice, 
+ *   this list of conditions and the following disclaimer. 
+ *
+ * - Redistributions in binary form must reproduce the above copyright notice, 
+ *   this list of conditions and the following disclaimer in the documentation 
+ *   and/or other materials provided with the distribution. 
+ *
+ * - Neither the name of the <ORGANIZATION> nor the names of its contributors 
+ *   may be used to endorse or promote products derived from this software without 
+ *   specific prior written permission. 
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND 
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED 
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. 
+ * IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, 
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, 
+ * OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, 
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY 
+ * OF SUCH DAMAGE. 
+ *
+ */
+
+#include	<stdio.h>
+#include	<fcntl.h>
+#include	<stdlib.h>
+#include	<iconv.h>
+#include	<locale.h>
+#include	<wchar.h>
+#include	<errno.h>
+#include	"odbc_portable.h"
+#include	"odbcinst.h"
+#include	"odbc_resource.h"
+#include	"odbc_connection.h"
+#include	"odbc_util.h"
+
+#ifndef OLECHAR
+typedef WCHAR OLECHAR;
+#endif
+
+#ifndef BSTR
+typedef OLECHAR *BSTR;
+#endif
+
+PUBLIC INT_PTR CALLBACK
+ConfigDSNDlgProc (HWND hwndParent, UINT message, WPARAM wParam,
```

Comment:
HWND Type is the type used by Windows.
On linux, a compile error seems to occur.

Reply:
I think actually, ConfigDSNDlgProc may not be called in Linux. And type HWND is defined as 'void *' in sqltypes.h (/usr/include) by unixODBC. No compiler error for **HWND** will be occurred even it is a meaningless code at the moment.